jung-algorithms/src/test/java/edu/uci/ics/jung/layout/spatial/FRLayoutsTimingTest.java

```diff
@@ -70,23 +74,10 @@ public void testFRBHVisitor() {
   }
 
   private void doTest(LayoutAlgorithm<String> layoutAlgorithm) {
-    layoutModel
-        .getLayoutStateChangeSupport()
-        .addLayoutStateChangeListener(
-            new LayoutModel.LayoutStateChangeListener() {
-              long time = System.currentTimeMillis();
-
-              @Override
-              public void layoutStateChanged(LayoutModel.LayoutStateChangeEvent evt) {
-                if (evt.active == false) {
-                  long endTime = System.currentTimeMillis();
-                  log.info(
-                      "elapsed time for {} was {}",
-                      layoutAlgorithm.getClass().getName(),
-                      endTime - time);
-                }
-              }
-            });
+    long startTime = System.currentTimeMillis();
     layoutModel.accept(layoutAlgorithm);
+    layoutModel
+        .getTheFuture()
+        .thenRun(() -> log.info("elapsed time {}", System.currentTimeMillis() - startTime));
   }
```

Comment:
Intended to replace the above sleep hack, and replace the burden of using event dispatches for timing alerts.